tests/test_supervisor_planner.py

```
@@ -660,6 +662,117 @@ async def test_db_error(self, planner: SupervisorPlanner) -> None:
+        p = SupervisorPlanner(config=cfg, project_dir=project_dir, session_factory=sf)
+        yield p
+        await close_db()
+
+    async def test_with_developer_runs(self, db_planner: SupervisorPlanner) -> None:
+        from datetime import datetime, timedelta, timezone
+
+        from sova.db.models import TaskRun
+
+        now = datetime.now(timezone.utc)
+        async with db_planner._session_factory() as session:
+            async with session.begin():
```

Ruff (0.16.1)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)